openai-core/src/commonMain/kotlin/com.aallam.openai.api/assistant/AssistantResponseFormat.kt

```diff
@@ -12,40 +20,56 @@ import kotlin.jvm.JvmInline
  * If text, the model can return text or any value needed.
  * type: string Must be one of text or json_object.
  */
-@Serializable
 @BetaOpenAI
-public sealed interface AssistantResponseFormat {
-
-    /**
-     * The type of response format being defined: text
-     */
-    @BetaOpenAI
+@Serializable(with = AssistantResponseFormat.ResponseFormatSerializer::class)
+public data class AssistantResponseFormat(
```

Comment:
I tried something [here](https://github.com/aallam/openai-kotlin/blob/feat/assistants-v2/openai-core/src/commonMain/kotlin/com.aallam.openai.api/assistant/AssistantResponseFormat.kt), can't remember if it works

Reply:
Yeah actually I started from your branch, and this one failed while deserializing. Tried several fixes, but I had to come up with custom serializer/deserializer. Wonder if simpler way is valid.